modules/store/src/models.ts

```diff
@@ -88,7 +94,7 @@ export type PropsReturnType<T extends object> = T extends any[]
  */
 export type ActionCreator<
   T extends string = string,
-  C extends Creator = Creator
+  C extends Creator = TypedCreator<T>
```

Comment:
The second generic type cannot be `Creator` since return type must contain `type` property. So we can either:
1. create new type (such as `TypedCreator`) or
2. have `TypePropertyIsNotAllowed` somehow generically toggleable

Looking forward to your opinion